akka-actor-typed-tests/src/test/scala/docs/akka/typed/supervision/SupervisionCompileOnly.scala

```diff
@@ -101,4 +104,30 @@ object SupervisionCompileOnly {
     }
   }
   //#restart-keep-children
+
+  trait Resource {
+    def close()
+    def process(parts: Array[String])
+  }
+  def claimResource(): Resource = ???
+  //#restart-PreRestart-signal
+  Behaviors.supervise {
+    Behaviors.setup { ctx =>
+      val resource = claimResource()
+
+      Behaviors
+        .receiveMessage[String] { msg =>
+          // there might be bugs here...
```

Comment:
what does this mean?

Reply:
that phrase is used in some other examples here, means that an exception might be thrown since the topic is about supervision and restarts

Reply:
Copy pasta from the previous restart example, just means the logic could throw unexpected failures (and you can infer one if there are no spaces in the string I guess).